src/gt4py/next/ffront/decorator.py

```diff
@@ -215,6 +215,13 @@ def itir(self) -> itir.FencilDefinition:
         return past_to_itir.PastToItirFactory()(no_args_past).program
 
     def __call__(self, *args, offset_provider: dict[str, Dimension], **kwargs: Any) -> None:
+        params = self.past_stage.past_node.params
```

Comment:
This will break the caching of the gtfn backend. Let's fix this together. Since this is such a performance critical part and for readability I would suggest to outsource the loop into a cached property that computes all the artifically added offset provider and only do a single
```suggestion
        offset_provider = offset_provider | self._implicit_offset_provider
```
here.